File: backend/lib/cache.py
"""Simple in-memory cache with 1-hour TTL — mirrors cache.ts"""
from datetime import datetime, timedelta
from typing import Any, Optional

CACHE_TTL = timedelta(hours=1)

_store: dict[str, dict] = {}


def get_cache(key: str) -> Optional[Any]:
    entry = _store.get(key)
    if entry is None:
        return None
    if datetime.now() - entry["timestamp"] > CACHE_TTL:
        del _store[key]
        return None
    return entry["data"]


def set_cache(key: str, data: Any) -> None:
    _store[key] = {"data": data, "timestamp": datetime.now()}

File: src/lib/cache.ts
// Simple in-memory cache with TTL (1 hour)
const CACHE_TTL_MS = 60 * 60 * 1000;

interface CacheEntry {
  data: unknown;
  timestamp: number;
}

const store = new Map<string, CacheEntry>();

export function getCache<T>(key: string): T | null {
  const entry = store.get(key);
  if (!entry) return null;
  if (Date.now() - entry.timestamp > CACHE_TTL_MS) {
    store.delete(key);
    return null;
  }
  return entry.data as T;
}

export function setCache(key: string, data: unknown): void {
  store.set(key, { data, timestamp: Date.now() });
}
